workspaces/cli-server/src/analytics.ts

```diff
@@ -0,0 +1,79 @@
+import { newAnalyticsEventBus } from '@useoptic/analytics/lib/eventbus';
+//@ts-ignore
+import Analytics from 'analytics-node';
+import { consistentAnonymousId } from '@useoptic/analytics/lib/consistentAnonymousId';
+import {
+  ClientContext,
+  TrackingEventBase,
+} from '@useoptic/analytics/lib/interfaces/TrackingEventBase';
+import { AnalyticsEventBus } from '@useoptic/analytics/lib/eventbus';
+import path from 'path';
+import os from 'os';
+import { IUserCredentials } from '@useoptic/cli-config';
+import fs from 'fs-extra';
+const packageJson = require('../package.json');
+
+const clientId = `local_cli_${packageJson.version}`;
+
+//event bus for tracking events
+const analyticsEvents: AnalyticsEventBus = newAnalyticsEventBus(
+  async (batchId: string) => {
+    const user = await getCredentials();
+    const clientAgent = user ? user.token : consistentAnonymousId;
+
+    const clientContext: ClientContext = {
+      clientAgent: clientAgent,
+      clientId: clientId,
+      clientSessionInstanceId: batchId,
```

Comment:
clientSessionInstanceId corresponds to the client lifecycle. For example, in a browser each tab would have a different one, and each cli run would have a different one. batchId is an additional field, like a correlation or causation id, helping to group events by their originating cause